creator-node/src/snapbackSM.js

```diff
@@ -449,71 +449,67 @@ class SnapbackSM {
   }
 
   /**
-   * Track an ongoing sync operation for a given secondaryUrl and user wallet
-   *  - Will re-enqueue sync if it fails or is still behind after retries or max duration
+   * Monitor an ongoing sync operation for a given secondaryUrl and user wallet
+   * Return boolean indicating if an additional sync is required
+   *
+   * Polls secondary for MaxSyncMonitoringDurationInMs
    */
-  async monitorSecondarySync (userWallet, primaryClockValue, secondaryUrl, syncType) {
-    const startTime = Date.now()
+  async additionalSyncIsRequired (userWallet, primaryClockValue, secondaryUrl, syncType) {
+    const MaxExportClockValueRange = this.nodeConfig.get('maxExportClockValueRange')
+    const logMsgString = `additionalSyncIsRequired (${syncType}): wallet ${userWallet} secondary ${secondaryUrl} primaryClock ${primaryClockValue}`
 
-    // Define axios request object for secondary sync status request
-    const syncMonitoringRequestParameters = {
+    // Define axios request object for secondary clock status request
+    const clockStatusRequestParams = {
       method: 'get',
       baseURL: secondaryUrl,
-      url: `/sync_status/${userWallet}`,
+      url: `/users/clock_status/${userWallet}`,
       responseType: 'json'
     }
 
-    // TODO syncAttemptCompleted does nothing - refactor this
-    let syncAttemptCompleted = false
-    let secondaryClockValAfterSync = null
-    while (!syncAttemptCompleted) {
+    const startTimeMs = Date.now()
+    const endTimeMs = startTimeMs + MaxSyncMonitoringDurationInMs
+
+    let additionalSyncRequired = true
+    while (Date.now() < endTimeMs) {
       try {
-        const syncMonitoringResp = await axios(syncMonitoringRequestParameters)
+        const clockStatusResp = await axios(clockStatusRequestParams)
+        const { clockValue: secondaryClockValue } = clockStatusResp.data.data
 
-        const respData = syncMonitoringResp.data.data
-        this.log(`processSync ${userWallet} secondary response: ${JSON.stringify(respData)}`)
+        this.log(`${logMsgString} secondaryClock ${secondaryClockValue}`)
 
-        // A success response does not necessarily guarantee completion, validate response data to confirm
-        // Returned secondaryClockValue can be greater than the cached primaryClockValue if a client write was initiated
-        //    after primaryClockValue cached and resulting sync is monitored
-        if (respData.clockValue >= primaryClockValue) {
-          secondaryClockValAfterSync = respData.clockValue
+        /**
+         * One sync op can process at most MaxExportClockValueRange range
+         * A larger clock diff will require multiple syncs; short-circuit monitoring
+         */
+        if (secondaryClockValue + MaxExportClockValueRange < primaryClockValue) {
```

Comment:
if we are adding a new secondary and primary clock is 4 *MaxExportClockValueRange , is there always an additional sync required? slash is that just okay bc it will be deduped if it is ongoing?

Reply:
yeah there is always an additional sync required as each sync will progress MaxExportClockValueRange. and yep it will be deduped, and even further - if it isn't deduped it will 429 since the previous one holds the lock. every scenario should be safe